server/resourcemanager/resource_manager.go

```diff
@@ -288,6 +297,15 @@ func (m *manager[T]) upsert(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if err := targetResource.Spec.Validate(); err != nil {
```

Comment:
this is already handled on `doCreate`, and if you move [this](https://github.com/kubeshop/tracetest/pull/2842/files#diff-73567862de3c0b65102c45d3df1f98a49f2649fe588140f55a56fa693068bf51R351) to happen within the `doUpdate` method, we can have both cases covered and remove the need for extra work here